omod/src/main/web/dashboardwidgets/obsacrossencounters/obsacrossencounters.controller.js

```diff
@@ -29,177 +28,117 @@ export default class ObsAcrossEncountersController {
+          for (let member of matchingMembers) {
+            matchingObsByUuid[member.concept.uuid].push(member);
+          }
         }
       }
-    }
-    if (lastShortName !== null && lastShortName.length > 0) {
-      concept.display = lastShortName;
-    }
-    return concept;
-  }
-
-
-  parseEncounters(encounters) {
-    angular.forEach(encounters, (encounter) => {
-      let searchObs = {};
-      angular.forEach(encounter.obs, (obs) => {
-        let conceptKeys = Object.keys(this.conceptsMap);
-
-        if (conceptKeys.findIndex(conceptKey => conceptKey === obs.concept.uuid) >= 0) {
-          //we found an obs match
-          searchObs[obs.concept.uuid] = obs;
-        }
-        if (obs.groupMembers != null && obs.groupMembers.length > 0) {
-          // need to search the groupMembers
-          let foundObs = this.parseGroupMembers(obs.groupMembers, conceptKeys);
-          if (typeof foundObs !== 'undefined' && foundObs !== null && Object.keys(foundObs).length > 0) {
-            if (Object.keys(foundObs).every(item => conceptKeys.includes(item))) {
-              //this is a complete pair of matching obs with a given concept uuid within the same obsGroup
-              let enc = {
-                encounterType: encounter.encounterType.name,
-                encounterDatetime: encounter.encounterDatetime,
-                obs: foundObs
-              };
-              this.simpleEncs.push(enc);
-            } else {
-              // we have an incomplete match, just add the values to the existing object
-              for (key in Object.keys(foundObs)){
-                searchObs[key] = searchObs[key];
-              }
-            }
+      // We create as many rows as the maximum number of times any obs
+      // in the encounter is recorded, packing obs into the first
+      // available row
+      let moreObsToShow = true;
+      while (moreObsToShow) {
+        moreObsToShow = false;
+        const obsRow = {};
+        for (let key of conceptKeys) {
+          const obs = matchingObsByUuid[key].shift();
+          if (obs) {
+            moreObsToShow = true;
+            obsRow[key] = obs;
           }
         }
-      });
-
-      if (Object.keys(searchObs).length > 0) {
-        let tempEnc = {
-          encounterType: encounter.encounterType.name,
-          encounterDatetime: encounter.encounterDatetime,
-          obs: searchObs
-        };
-        this.simpleEncs.push(tempEnc);
+        if (moreObsToShow) {
+          this.simpleEncs.push({
+            encounterType: encounter.encounterType.name,
+            encounterDatetime: encounter.encounterDatetime,
+            obs: obsRow
+          });
+        }
       }
       // if the widget was configured to display the obs concept SHORT name instead of the default obs.value.display value
       // By default, the widget displays the obs.value.display property
-      if (this.config.useConceptShortName && this.config.useConceptShortName === 'true') {
```

Comment:
This is a functional change. Previously `useConceptShortName` only worked if provided the string value `"true"`. It now works for `true`. It will still work for `"true"`, so it isn't a breaking change (unless anyone specified the string `"false"`, but that seems unlikely).